### PR TITLE
Feat: add dex addon for velaux

### PR DESCRIPTION
**Pull request by FogDong**
Signed-off-by: FogDong <dongtianxin.tx@alibaba-inc.com>

### Checklist

I have:

- [ ] Title of the PR starts with type (e.g. `[Addon]` or `[Trait]`).
- [ ] Updated/Added any relevant [documentation] and [examples].
- [ ] Unit/E2E Tests added.

/cc @barnettZQG

### CODE REVIEW

File: addons/dex/readme.md
Comment (wonderflow) [inline addons/dex/readme.md:3]:
```suggestion
This addon provides dex login for VelaUX.

Dex is an identity service that uses [OpenID Connect ](https://openid.net/connect/)to drive authentication for other apps.

Dex acts as a portal to other identity providers through [“connectors.”](https://dexidp.io/docs/connectors/) This lets Dex defer authentication to LDAP servers, SAML providers, or established identity providers like GitHub, Google, and Active Directory. Clients write their authentication logic once to talk to Dex, then Dex handles the protocols for a given backend.

Please refer to [Dex website](https://dexidp.io/docs/) for more details.
```

**Review by wonderflow — APPROVED** [review]
LGTM after the readme nits

**Review by wonderflow — DISMISSED** [review]

**Review by wonderflow — APPROVED** [review]

**Comment by FogDong** [comment]
Hold this until we can manage the Dex Connectors in apiserver.

**Review by wangyikewxgm — DISMISSED** [review]

**Review by wonderflow — DISMISSED** [review]

File: addons/dex/resources/dex-config-app.cue
Comment (barnettZQG) [inline addons/dex/resources/dex-config-app.cue:28]: What data is dex need storage?

Comment (FogDong) [reply addons/dex/resources/dex-config-app.cue:28]: Dex requires persisting state to perform various tasks such as track refresh tokens, preventing replays, and rotating keys.

File: addons/dex/deifnitions/dex-connectors-def.yaml
Comment (zzxwill) [inline addons/dex/deifnitions/dex-connectors-def.yaml:1]: Addon can identify a .cue. So it might be better to keep the .cue file here.

Comment (FogDong) [reply addons/dex/deifnitions/dex-connectors-def.yaml:1]: For addons, all definition is `YAML` file in the directory.
If we want to change it, it's better that we change it all.

File: addons/dex/metadata.yaml
Comment (barnettZQG) [inline addons/dex/metadata.yaml:2]: use the dex version?

**Review by wangyikewxgm — APPROVED** [review]

**Pull request merged**
